ale_linters/sh/shellcheck.vim

```diff
@@ -16,9 +16,21 @@ else
     let s:exclude_option = ''
 endif
 
+function! s:GetDialectArgument()
+    if exists('b:is_bash') && b:is_bash
+        return '-s bash'
+    elseif exists('b:is_sh') && b:is_sh
+        return '-s sh'
+    elseif exists('b:is_kornshell') && b:is_kornshell
+        return '-s ksh'
+    endif
```

Comment:
You could probably include zsh here too. 

I didn't know about these special buffer variables.

Could you use `command_callback` instead below with a function which returns the command? Then you could re-evaluate the command each time ALE runs the linter. This here will only be evaluated once.